### PR TITLE
Deprecation of mt_rand in uuid generation

### PR DESCRIPTION
**Pull request by nojimage**
fixes for #13944

### CODE REVIEW

**Comment by rchavik** [comment]
This adds a dependency and also throw a depreciationWarning.

It should be one OR the other.

File: src/Utility/Text.php [hunk 1]
```diff
@@ -61,6 +61,9 @@ class Text
     public static function uuid()
     {
         $random = function_exists('random_int') ? 'random_int' : 'mt_rand';
```
Comment (rchavik) [inline src/Utility/Text.php]: if random_compat is installed, we can immediately do: 

```php
$random = 'random_int';
```

am i missing something?

Comment (nojimage) [reply src/Utility/Text.php]: When upgrading an existing application, there seems to has a short time random_int to not exist, between updating the package with composer.
If we can ignore this, we can unify to random_int.

**Comment by dereuromark** [comment]
I would recommend letting people opt in rather than requiring this.

File: composer.json [hunk 2]
```diff
@@ -35,7 +35,8 @@
         "aura/intl": "^3.0.0",
         "psr/log": "^1.0.0",
         "psr/simple-cache": "^1.0.0",
-        "zendframework/zend-diactoros": "^1.4.0"
+        "zendframework/zend-diactoros": "^1.4.0",
+        "paragonie/random_compat": "^2.0|9.99.99"
```
Comment (othercorey) [inline composer.json]: What does requiring 9.99.99 do?

Comment (garas) [reply composer.json]: PHP5 will install v2.0 with `lib/random.php` autoloaded.

PHP7 will install v9.99.99 with no file autoloaded.

Comment (othercorey) [reply composer.json]: I see now. I was looking in master branch.  Very nice, so this solves our problem.

File: src/Utility/Text.php (hunk 1 above)
Comment (othercorey) [reply src/Utility/Text.php]: > When upgrading an existing application, there seems to has a short time random_int to not exist, between updating the package with composer.

The composer.json change makes that impossible though, right?

**Comment by othercorey** [comment]
I think 3.9 is probably better for this.  We can add it to migration notes.

**Comment by othercorey** [comment]
> I would recommend letting people opt in rather than requiring this.

symfony seems to use this in their polyfill for php7

File: src/Utility/Text.php (hunk 1 above)
Comment (othercorey) [reply src/Utility/Text.php]: I removed the deprecation notice and defaulted to `random_int()`. If PHP suggests this implementation in their official doc, it's good enough for me.

https://www.php.net/manual/en/function.random-int.php

**Comment by othercorey** [comment]
I'm not sure why it is failing to run.  I'll have to clone locally.

**Comment by nojimage** [comment]
> I'm not sure why it is failing to run. I'll have to clone locally.

It seems to polyfilled `Error` class in ramdom_compat. but `ParseError` is not.
 https://github.com/paragonie/random_compat/blob/9e071856f328b3d249df75b3256394d398809ed7/lib/error_polyfill.php#L29

**Comment by othercorey** [comment]
> > I'm not sure why it is failing to run. I'll have to clone locally.
> 
> It seems to polyfilled `Error` class in ramdom_compat. but `ParseError` is not.
> https://github.com/paragonie/random_compat/blob/9e071856f328b3d249df75b3256394d398809ed7/lib/error_polyfill.php#L29

I was just looking at why Error was defined, thanks!  I fixed it with a proper version check like everywhere else.

**Comment by othercorey** [comment]
I made the PR list all messy by messing with 3.next trying to find the build break.  Doesn't matter for the merge though.

**Review by ADmad — APPROVED** [review]
Changes look good but there's lot of commit noise. Please squash them into a single commit.

**Comment by othercorey** [comment]
> Changes look good but there's lot of commit noise. Please squash them into a single commit.

Done.

**Pull request merged**